apps/golf-gps/golf-gps.js

```diff
@@ -0,0 +1,306 @@
+    46,
+    atob("FBkbFBsaHBobGxsbEw=="),
+    48 | 65536
+  );
+};
+
+var courseData = require("Storage").open("course-data", "r");
+
+var lastFix = {
+  lat: 0,
+  lon: 0,
+  alt: 0, // altitude in m
+  speed: 0, // km/h
+  course: 0, // heading in degrees
+  time: 0,
+  satellites: 0,
+  fix: 0,
+  hdop: 0, // x5 ~ meter accuracy
+};
+
+const zeroPad = (num, places) => String(num).padStart(places, '0');
+
+function onGPS(fix) {
+  Object.assign(lastFix, fix);
+  if (lastFix.fix && playON) showPlayData();
+}
+
+function radians(degrees) {
+  return degrees * Math.PI / 180;
+}
+
+function readOneCourseData() {
+  // Clear previous data
+  lat = [];
+  lon = [];
+  par = [];
+  let l = courseData.readLine();
+  courseName = l;
+  // Read one course data & initialize score
+  for (let i = 1; i < 19; i++) {
+    l = courseData.readLine();
+    if (l !== undefined) {
+      const parts = l.split(',');
+      lat[i] = parseFloat(parts[0]);
+      lon[i] = parseFloat(parts[1]);
+      par[i] = parseInt(parts[2]);
+      score[i] = 0;
+    }
+  }
+}
+
+function distanceCalc(lat1, long1, lat2, long2) {
+  let delLat = Math.abs(lat1 - lat2) * 111194.9; // 111194.9 = (2*6371000*pi)/360, 6371000 ~ Earth's average radius
+  let delLong = Math.abs(long1 - long2) * 111194.9 * Math.cos(radians((lat1+lat2)/2));
+  return Math.sqrt(delLat * delLat + delLong * delLong)/0.9144; // in yards
+}
+
+function mainMenu() {
+  E.showPrompt("Play now or\n\nView scores?", {
+    title: "Golf GPS",
+    buttons: {
+      "PLAY": 1,
+      "VIEW": 2
+    }
+  }).then(choice => {
+    if (choice === 1) fixGPS();
+    else browseScore();
+  });
+}
+
+function browseScore() {
+  const scoreFiles = require("Storage").list(/^Scorecard-/);
+  if (scoreFiles.length === 0) {
+    E.showPrompt("No score file found.\n\nYou need to play at least a game.", {
+      title: `Error`,
+      buttons: {
+        "END": 1
+      }
+    }).then(choice => {
+      if (choice === 1) load();
+    });
+  }
```

Comment:
Do you need to `return` after issuing this `E.showPrompt()` ? Otherwise I expect the below browse-files prompt will be shown

Reply:
Well, there is only one button to 'END', so it will go back to the clock when you tap on it. I have tested this case and it worked fine. Any thoughts?

Reply:
I might have mangled my question a little - what I'm wondering is:
```diff
 function browseScore() {
   const scoreFiles = require("Storage").list(/^Scorecard-/);
   if (scoreFiles.length === 0) {
     E.showPrompt("No score file found.\n\nYou need to play at least a game.", {
       ...
     }).then(choice => {
       if (choice === 1) load();
     });
 
+    // we get to here after having shown the prompt above
   }
   let fileIndx = scoreFiles.length - 1;
 
   function browseFiles() {
     const browsefile = require("Storage").open(scoreFiles[fileIndx].substring(0, 18), "r");
     const l = browsefile.read(80);
 
     E.showPrompt(l, {
       buttons: {
         "<<": 1,
         ">>": 2,
         "End": 3
       }
     }).then(choice => {
       if (choice === 1) fileIndx = (fileIndx - 1 + scoreFiles.length) % scoreFiles.length;
       else if (choice === 2) fileIndx = (fileIndx + 1) % scoreFiles.length;
       else if (choice === 3) load();
       browseFiles();
     });
   }
 
+  // and so we end up down here, and we'll call browseFiles(),
+  // which will show another prompt - I'd expect this will overwrite the first prompt
   browseFiles();
 }
```

Reply:
Ah... I see what you meant. Yes, you're right. It should look like this.

```
   }
   let fileIndx = scoreFiles.length - 1;
   return 0;
   function browseFiles() {
```
Thanks.

Reply:
I'm afraid not - the prompt is asynchronous so the callbacks will run quite a while after the function's finished :)

Reply:
Interesting that it didn't overwrite - I should have some time to take a look next week if you don't get to the bottom of it in the meanwhile, or if it's not an issue we can close this

(there's just [the final comment](https://github.com/espruino/BangleApps/pull/3467#discussion_r1664728301) below to deal with)

Reply:
Oh, I see. I will see if I can find some time to take a look today.
Thanks.

Reply:
I have separated the functions, browseScore() and browseFiles() and changed the variables fileIndx and scoreFiles to global variable. I'm not sure whether this is good solution or not, but with this change, the prompts still works fine and don't see any error in the console.  @bobrippling, please review when you have time. Thanks.

Reply:
Sounds good, glad to hear the prompt is working still and thanks for your changes!